lib/utils.pm

```diff
@@ -1980,6 +1984,8 @@ sub install_patterns {
         if (($pt =~ /sap_server/) && is_sle('=11-SP4')) {
             next;
         }
+        # For Public cloud module test we need install 'Tools' but not 'Instance' pattern if outside of public cloud images.
+        next if (($pt =~ /OpenStack/) && ($pt !~ /Tools/) && check_var('PUBLIC_CLOUD_IMG', '1'));
```

Comment:
can you give examples of strings which you do want to match and examples of strings which you want to skip ? I think it will look more clear and will work better as single regex ...

Reply:
There are 4 patterns for openstack:
openstack
openstack-instance-init
openstack-instance-tools
openstack-tools

For non public images env, we only install 'openstack-tools', here need skip 'openstack'.